docs/proposals/hpa/centralized-hpa.md

```diff
@@ -0,0 +1,181 @@
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+As a user, I have HPA in each cluster for one application. Since it's quite complex and redundant to manage them separately, I expect to be able to config and manage the HPA resource in a unified manner.
+
+#### Story 2
+As a user, I deploy my application in multiple clusters. Since the application's load may change with the incoming requests, I expect to be able to scale up or down the workloads across clusters.  
+PS: After scaling up or down, PropagationPolicy will determine where the replicas will be scheduled.
+
+#### Story 3
+As a user, I deploy my service with HPA in a single cluster. Since this cluster may scaling fail becuase of single cluster limitation, such as resource limitations, which will cause the service unavailable if requests too many, so I expect to improve the scaling stability of the service with scaling across clusters.
+
+#### Story 4
+As a user, I deploy my application with HPA in multiple clusters. Since the unlimited scaling up of the application may lead to unexpected cloud costs and resource congestion that may cause service disruptions for other applications, so I expect to set the maximum number of instances for multi-cluster applications.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go into as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+1. The workloads/pods in different member clusters selected by the same HPA CR/resource share the load of the application equally. For example, 10 pods of the application are spread into two member clusters with distribution `cluster1: 3 pods, cluster2: 7 pods`, so the 3 pods in cluster1 take 3/10 of total requests and 7 pods in cluster2 take 7/10 of total requests. Scenarios that don't meet the restriction are not considered in this proposal.
+1. If the system does not meet the first requirement, some clusters' services may become overloaded while others remain underloaded.      
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate them? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+1. If the HPA is propagated to member clusters, there will be conflicts with Karmada control plane. So users couldn't propagate HPA from Karmada control plane.
+2. The HPA used in a single cluster can be seamlessly migrated to multiple clusters' centralized HPA.
+
+## Design Details
+### Architecture
+For different scenarios, the architecture may be different:  
+HPA with CPU/Memory metrics:  
+![K8s native metrics](./statics/centralizedhpa-native-metrics.png)  
```

Comment:
Does nativew metrics server support multiple clusters? How does metrics server know which apiserver to request when HPA controller start a new request?

Reply:
> Does nativew metrics server support multiple clusters?
From the code of K8s native HPA controller: https://github.com/kubernetes/kubernetes/blob/9db01b6eb32995b3d0bc76e3693ebcee80ef0a82/pkg/controller/podautoscaler/replica_calculator.go#L66

The responsibility of Karmada-metrics-server is to query the metrics from member clusters and do aggregation.
So it hasn't got connection with native metrics server.

> How does metrics server know which apiserver to request when HPA controller start a new request?   

From RB